capa/features/extractors/vmray/call.py

```diff
@@ -2,62 +2,115 @@
 # Licensed under the Apache License, Version 2.0 (the "License");
 #  you may not use this file except in compliance with the License.
 # You may obtain a copy of the License at: [package root]/LICENSE.txt
-# Unless required by applicable law or agreed to in writing, software distributed under the License
-#  is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# Unless required by applicable law or agreed to in writing
+# software distributed under the License
+#  is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND
+# , either express or implied.
 # See the License for the specific language governing permissions and limitations under the License.
+
 import logging
 from typing import Tuple, Iterator
 
 from capa.helpers import assert_never
 from capa.features.insn import API, Number
 from capa.features.common import String, Feature
 from capa.features.address import Address
+from capa.features.extractors.vmray.models import FunctionCall
 from capa.features.extractors.base_extractor import CallHandle, ThreadHandle, ProcessHandle
 
 logger = logging.getLogger(__name__)
 
 
-def extract_call_features(ph: ProcessHandle, th: ThreadHandle, ch: CallHandle) -> Iterator[Tuple[Feature, Address]]:
+def extract_function_calls(ph: ProcessHandle, th: ThreadHandle, ch: CallHandle) -> Iterator[Tuple[Feature, Address]]:
     """
-    this method extracts the given call's features (such as API name and arguments),
-    and returns them as API, Number, and String features.
+    This method extracts the given call's features (such as API name and
+    arguments), and returns them as API, Number, and String features.
 
-    args:
-      call: FunctionCall object representing the XML fncall element
+    Args:
+        ph: process handle (for defining the extraction scope)
+        th: thread handle (for defining the extraction scope)
+        ch: call handle (for defining the extraction scope)
 
-      yields: Feature, address; where Feature is either: API, Number, or String.
+    Yields:
+        Feature, address; where Feature is either: API, Number, or String.
     """
 
-    # TODO (meh): update for new models https://github.com/mandiant/capa/issues/2148
-    # print(ch)
-    return
+    # TODO(meh): update for new models
+    # https://github.com/mandiant/capa/issues/2148
 
-    # Extract API name
-    yield API(ch.inner.name), ch.inner.address
+    call: FunctionCall = ch.inner
 
     # Extract arguments from <in>
-    for param in ch.inner.in_:
-        value = param.value
-        if isinstance(value, str):
-            yield String(value), ch.inner.address
-
-        elif isinstance(value, int):
-            yield Number(value), ch.inner.address
 
-        else:
-            assert_never(value)
+    if call is not None:
+        for param in call.in_:
```

Comment:
re:mypy:```capa/features/extractors/vmray/call.py:46: error: Item "None" of "Optional[In_Out]" has no attribute "__iter__" (not iterable)  [union-attr]```

`in_` is declared as `Optional`: https://github.com/mandiant/capa/blob/45014e32cdd186408f12996b067f43f113037f54/capa/features/extractors/vmray/models.py#L35

Because `in_` is `Optional` i.e. "can be `None`" we need to check if `in_` is `None` before attempting to user it as an iterator. `mypy` is helping us catch this `None` case before we encounter it at execution (thanks `mypy` 😄 ) which would throw an exception